### PR TITLE
Extra Functionality for CvarComboBox

### PR DESCRIPTION
**Pull request by braem**
Related to #592 

In converting cvar-tied comboboxes to cvarcombobox, I found that it did not have the needed functionality to use it on cvars with minimum values other than 0. To do this I 

1. Added `GetMin` and `GetMax` functions to `ConVarRef`.
2. Used `GetMin` to change how the combobox indexes.
3. Used `GetMax` to solve an issue where the combobox was not setting a value in some cases.
    - The `Reset()` call in the constructor was removed as it didn't do anything since the items have not been added to the combobox yet.
    - Added an `AddItem` override to call `Reset()` when all the items have been added to the combobox.

After this was done, the comboboxes in the momentum settings pages were converted to use this. 

The GameUI comboboxes were not changed as the current implementation `CvarComboBox` does not support

1. Cvars that have non-linear values, such as `mat_forceaniso`.
2. Comboboxes that set multiple cvars.
    - eg. Filtering mode combobox sets `mat_trilinear` or `mat_forceaniso` depending on the active item.

I believe `CvarComboBox` will need more functionality, perhaps for a different PR as this is mainly converting the momentum settings pages.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/vgui2/vgui_controls/CvarComboBox.cpp [hunk 1]
```diff
@@ -20,6 +20,18 @@ CvarComboBox::CvarComboBox(Panel *parent, const char *panelName, int numLines, b
 {
     InitSettings();
     m_bIgnoreMissingCvar = ignoreMissingCvar;
+    float m_flCvarMin, m_flCvarMax;
+    m_iCvarMin = m_cvar.GetMin(m_flCvarMin) ? RoundFloatToInt(m_flCvarMin) : 0;
+    m_iCvarMax = m_cvar.GetMin(m_flCvarMax) ? RoundFloatToInt(m_flCvarMax) : 0;
+    float m_flCvarMin;
```
Comment (Gocnak) [inline mp/src/vgui2/vgui_controls/CvarComboBox.cpp]: Only use `m_` for member variables. You also seem to be declaring `m_flCvarMin` twice here.

Comment (braem) [reply mp/src/vgui2/vgui_controls/CvarComboBox.cpp]: Woops! All this should do is needlessly call `Reset()` anyway. Will fix this up.

**Comment by Gocnak** [comment]
Can you pull the converting out to its own branch based off this one? You also should look into the options panels in GameUI and see if they can be converted over as well.

**Comment by braem** [comment]
> 
> 
> Can you pull the converting out to its own branch based off this one? You also should look into the options panels in GameUI and see if they can be converted over as well.

Moved the momentum setting page's conversions to a new PR, #668 . The GameUI panel will require new functionality that this implementation of `CvarComboBox` does not have. Particularly cvars with non-linear values and comboboxes that set multiple cvars are not accounted for.

I believe those can be separated into a new PR, if you wish to continue converting those ones as well.

**Comment by Gocnak** [comment]
Nah don't bother, it'd be a bit of structuring in the res file, and while it is possible, it's not needed currently.

**Pull request merged**